src/python/pants/backend/adhoc/run_system_binary_integration_test.py

```diff
@@ -91,3 +91,35 @@ def test_fingerprint(fingerprint: str, passes: bool) -> None:
         else:
             assert result.exit_code != 0
             assert "Could not find a binary with name `bash`" in result.stderr.strip()
+
+
+def test_runnable_dependencies() -> None:
+    sources = {
+        "src/BUILD": dedent(
+            """\
+            system_binary(
+                name="bash",
+                binary_name="bash",
+            )
+
+            adhoc_tool(
+                name="adhoc",
+                runnable=":bash",
+                runnable_dependencies=[":bash",],
```

Comment:
I find this test slightly confusing when the same target is used both for the runnable tool and the runnable dependency. It is not immediately obvious what is going on, or why we would need to provide the same target twice, until I realized what's going on.

I recall having this confusion during my previous review as well, and from then to now I had to rediscover this conclusion so for me it's a real thing.

A comment explaining what's what may be enough.

Reply:
Good catch. I think it may be more intuitive if I invoke a different tool.

Reply:
This test now invokes `awk` as its subprocess. It's much clearer what's going on now.